src/middleware/report.validation.js

```diff
@@ -0,0 +1,149 @@
+            404,
+            'The article with this id is not found'
+          );
+        }
+        return next();
+      })
+      .catch(() => {
+        return responseGenerator.sendError(
+          res,
+          500,
+          'There is an internal server error'
+        );
+      });
+  },
+
+  /**
+   * @description Validate the violation type passed
+   * @param  {object} req The HTTP request object
+   * @param  {object} res The HTTP response object
+   * @param  {object} next The next middleware
+   * @returns {object} The response object or the next middleware
+   */
+  validateViolation: (req, res, next) => {
+    const violations = [
+      'Discrimination',
+      'Sexual Content',
+      'Offensive Language',
+      'Plagiarism'
+    ];
+
+    // Check if violation specified is allowed
+    const found = violations.find(element => element === req.body.violation);
+
+    if (!found) {
+      return responseGenerator.sendError(
+        res,
+        400,
+        `Invalid violation. Violation can be ['Discrimination', 'Plagiarism', 'Sexual Content', 'Offensive Language']`
+      );
+    }
+
+    return next();
+  },
+
+  /**
+   * @description Validate the already reported article
+   * @param  {object} req The HTTP request object
+   * @param  {object} res The HTTP response object
+   * @param  {object} next The next middleware
+   * @returns {object} The response object or the next middleware
+   */
+  validateAlreadyReported: async (req, res, next) => {
+    const fields = {
+      userId: req.currentUser.id,
+      articleId: req.params.articleId
+    };
+    const alreadyReported = await BaseRepository.findOneByField(
+      db.Cases,
+      fields
+    );
+
+    if (alreadyReported !== null) {
+      const { userId, articleId } = alreadyReported.dataValues;
+      if (
+        Number(userId) === Number(fields.userId) &&
+        Number(articleId) === Number(fields.articleId)
+      ) {
+        return responseGenerator.sendError(
+          res,
+          409,
+          'You have reported this article already'
+        );
+      }
+    }
+    return next();
+  },
+
+  /**
+   * @description Validate the objects passed in the body
+   * @param  {object} req The HTTP request object
+   * @param  {object} res The HTTP response object
+   * @param  {object} next The next middleware
+   * @returns {object} The HTTP response object or the next middleware
+   */
+  validateRequestObject: (req, res, next) => {
+    const { valid, invalidMessages } = checkProps(
+      req.body,
+      'violation',
+      'description'
+    );
+
+    if (!valid) {
+      return responseGenerator.sendError(res, 400, invalidMessages);
+    }
+
+    return next();
+  },
+  /**
+   * @param  {object} req Http Request object
+   * @param  {object} res Http Response object
+   * @param  {function} next Calls the next middleware
+   * @returns  {object} undefined
+   */
+  validateComments: (req, res, next) => {
+    const { content } = req.body;
```

Comment:
Change `content` to `body` because it is no longer part of the attributes of an article.

Reply:
yes, this is a standalone body params its has nothing to do with the article body but the content of the report.